src/main/java/com/aphyr/riemann/client/Promise.java

```diff
@@ -75,6 +99,11 @@ public Object unsafeDeref(final long time,
     }
   }
 
+  @Override
+  public void setCallback(Callback<T> c) {
```

Comment:
Perhaps I'm just not understanding, but Netty 3.6 only seems to have ChannelFutures, which doesn't quite seem like a fit (NB: I know nothing about Netty). Netty 4 has something that looks perfect: http://netty.io/4.0/api/io/netty/util/concurrent/DefaultPromise.html, but I'm guessing there's a good reason this is still on Netty 3.